definitions-trimmed-migrate/src/app.ts

```diff
@@ -27,15 +25,10 @@ export const app = async (config: AppConfig, logService: LogService) => {
         await mongoService.connect();
         await redisService.connect();
 
-        await migrationService.start(config.migrationOptions.process as MigrationServiceProcess);
+        await migrationService.start(config.migrationOptions);
     } catch (err) {
         logService.error('App.Exception', err, { exception: err as Error });
-    } finally {
-        await mongoService.disconnect();
-        await redisService.disconnect();
 
-        logService.log('App', `Migration completed in ${(Date.now() - startTime) / 1000} seconds`);
-
-        process.kill(process.pid, 'SIGINT');
+        await app(config, logService);
```

Comment:
Is this retry upon exception?  should we put a maxRetryTimes limit?

Reply:
this is just a 1-off super rare retry that i've only seen happen once